File: MergeTwoSortedLists/MergeTwoSortedLists.cpp
/**
* Definition for singly-linked list.
*/
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
public:
	ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
		if (l1 == nullptr && l2 == nullptr)
			return nullptr;
		ListNode pre_newhead(0);
		ListNode* node = &pre_newhead;
		while (l1 != nullptr && l2 != nullptr) {
			if (l1->val <= l2->val) {
				node->next = l1;
				l1 = l1->next;
			}
			else {
				node->next = l2;
				l2 = l2->next;
			}
			node = node->next;
		}
		if (l1 == nullptr)
			node->next = l2;
		else
			node->next = l1;
		return pre_newhead.next;
	}
};

File: MergeTwoSortedLists/MergeTwoSortedLists.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if l1 == None and l2 == None:
            return None
        if l1 == None and l2 != None:
            return l2
        if l1 != None and l2 == None:
            return l1
        tmp1 = l1
        tmp2 = l2
        if l1.val < l2.val:
            l3 = ListNode(l1.val)
            tmp1 = l1.next
        else:
            l3 = ListNode(l2.val)
            tmp2 = l2.next
        tmp3 = l3
        while tmp1 != None and tmp2 != None:
            if tmp1.val < tmp2.val:
                tmp3.next = tmp1
                tmp1 = tmp1.next
                tmp3 = tmp3.next
            else:
                tmp3.next = tmp2
                tmp2 = tmp2.next
                tmp3 = tmp3.next
        if tmp1 == None and tmp2 != None:
            tmp3.next = tmp2
            return l3
        if tmp1 != None and tmp2 == None:
            tmp3.next = tmp1
            return l3
        return l3